packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java

```diff
@@ -27,82 +100,132 @@
   private static final int SOURCE_CAMERA = 0;
   private static final int SOURCE_GALLERY = 1;
 
-  private final PluginRegistry.Registrar registrar;
+  private MethodChannel channel;
   private ImagePickerDelegate delegate;
-  private Application.ActivityLifecycleCallbacks activityLifecycleCallbacks;
+  private FlutterPluginBinding pluginBinding;
+  private ActivityPluginBinding activityBinding;
+  private Application applicationContext;
```

Comment:
nit: An `Application` and an `ApplicationContext` are two separate things, so this naming is a little confusing. Do you really need this to be an `Application`, or are you just trying to get an `ApplicationContext`? If it's the latter I would just stick with a `Context` for the type here.